src/shogun/classifier/AveragedPerceptron.h

```diff
@@ -69,8 +69,10 @@ namespace shogun
 		/** registers and initializes parameters */
 		void init();
 
-		virtual void init_model(std::shared_ptr<Features> data);
-		virtual void iteration();
+		void init_model(const std::shared_ptr<Features>& data) override;
+		virtual void iteration(
```

Comment:
since you are fixing these
```suggestion
		void iteration(
```